src/main/java/de/tum/in/www1/artemis/service/learningpath/LearningPathService.java

```diff
@@ -334,6 +335,9 @@ public NgxLearningPathDTO generateNgxPathRepresentation(@NotNull LearningPath le
      */
     public LearningPath findWithCompetenciesAndLearningObjectsAndCompletedUsersById(long learningPathId) {
         LearningPath learningPath = learningPathRepository.findWithCompetenciesAndLectureUnitsAndExercisesByIdElseThrow(learningPathId);
+        // Remove exercise units, since they are already retrieved as exercises
+        learningPath.getCompetencies().stream().forEach(competency -> competency
+                .setLectureUnits(competency.getLectureUnits().stream().filter(lectureUnit -> !(lectureUnit instanceof ExerciseUnit)).collect(Collectors.toSet())));
```

Comment:
The following implementation should be more robust for future changes 

```suggestion
        // Remove exercise units, since they are already retrieved as exercises
        learningPath.getCompetencies().stream().forEach(competency -> competency
                .setLectureUnits(competency.getLectureUnits().stream().filter(lectureUnit -> lectureUnit instanceof LectureUnit).collect(Collectors.toSet())));
```

Reply:
`LectureUnit` is the superclass of all possibilities, including `ExerciseUnit`